[py] Log complete python errors + tracebacks when check import fails

olivielpeau opened the pull request:
### What does this PR do?

In some cases the python module of a check can't be imported because
of a python code issue, in that case logging the traceback is useful. This PR
does just that.

### Motivation

Without this change it's impossible to debug an issue with the import of a python
check module, based on the logs alone.

### Additional Notes

Use a common function that retrieves and formats the python
interpreter error, and clears the error flag in the interpreter.
Any python c api call that returns an error should be followed by
a call of that common function to get a formatted string of the error.

The signature of that function could be changed at some point to return
something more meaningful than just a string, but I didn't feel the need for that
right now.

olivielpeau committed:
[py] Log complete python errors + tracebacks when check import fails

In some cases the python module of a check can't be imported because
of a python code issue, in that case logging the traceback is useful.

Use a common function that retrieves and formats the python
interpreter error, and clears the error flag in the interpreter.
Any python c api call that returns an error should be followed by
a call of that common function to get a formatted string of the error.

truthbk reviewed (changes requested):
Let's remove the nested lock for now, until we find an elegant way of dealing with this. We should also create an issue to look into this carefully.

olivielpeau committed:
[py] Don't use a nested StickyLock

Using a nested lock unlocks to goroutine to the thread when the child
lock is unlocked (whereas we'd want that to happen when the parent lock
is unlocked).

Until we can figure out a good way to deal with this situation, don't
use nested locks, and add a warning to `getPythonError`.
